File: Project_Euler/C++/run9.c
//Problem #9
//Find a*b*c such that a+b+c = 1000 and a^2 + b^2 = c^2

#include "/media/SW_Preload/Root/Georgia_Tech/BlackBox/basic.c"

int main()
{
  double alpha = 1000;
  double a,b,c;
  int i;
  for(i = 1;i<=1000;i++)
    {
      b = i;
      c = (2*alpha*b-2*SQUARE(b)-SQUARE(alpha))/(2*b-2*alpha);
      if(c > 0 && floor(c) == c)
	{
	  if (SQUARE(c) >= SQUARE(b))
	    {
	      a = sqrt(SQUARE(c) - SQUARE(b));
	      if(floor(a) == a) 
		{
		  scalardisp(a*b*c,"answer");
		  return 0;
		}
	    }
	}
    }
}

File: Project_Euler/MATLAB/run9.m
%Problem 9
%%Find a*b*c such that a+b+c = 1000 and a^2 + b^2 = c^2

tic

alpha = 1000;
for b = 1:1000
  c = (2*alpha*b-2*b^2-alpha^2)/(2*b-2*alpha);
  if c > 0 && round(c) == c && isreal(c)
    a = sqrt(c^2 - b^2);
    if a > 0 && round(a) == a && isreal(a)
      answer = a*b*c
      toc
      pause
    end
  end
end
